api/src/main/scala/hmda/api/http/InstitutionsHttpApi.scala

```diff
@@ -0,0 +1,200 @@
+  implicit val materializer: ActorMaterializer
+  val log: LoggingAdapter
+
+  implicit val timeout: Timeout
+
+  val splitLines = Framing.delimiter(ByteString("\n"), 2048, allowTruncation = true)
+
+  val institutionsPath =
+    path("institutions") {
+      val institutionsActor = system.actorSelection("/user/institutions")
+      get {
+        val fInstitutions = (institutionsActor ? GetState).mapTo[Set[Institution]]
+        onComplete(fInstitutions) {
+          case Success(institutions) =>
+            complete(ToResponseMarshallable(Institutions(institutions)))
+          case Failure(error) =>
+            log.error(error.getLocalizedMessage)
+            complete(HttpResponse(StatusCodes.InternalServerError))
+        }
+      }
+    }
+
+  val institutionByIdPath =
+    path("institutions" / Segment) { fid =>
+      extractExecutionContext { executor =>
+        val institutionsActor = system.actorSelection("/user/institutions")
+        val filingsActor = system.actorOf(FilingPersistence.props(fid))
+        get {
+          implicit val ec: ExecutionContext = executor
+          val fInstitutionDetails = institutionDetails(fid, institutionsActor, filingsActor)
+          onComplete(fInstitutionDetails) {
+            case Success(institution) =>
+              filingsActor ! Shutdown
+              complete(ToResponseMarshallable(institution))
+            case Failure(error) =>
+              filingsActor ! Shutdown
+              log.error(error.getLocalizedMessage)
+              complete(HttpResponse(StatusCodes.InternalServerError))
+          }
+        }
+      }
+    }
+
+  val filingByPeriodPath =
+    path("institutions" / Segment / "filings" / Segment) { (fid, period) =>
+      extractExecutionContext { executor =>
+        val filingsActor = system.actorOf(FilingPersistence.props(fid))
+        val submissionActor = system.actorOf(SubmissionPersistence.props(fid, period))
+        get {
+          implicit val ec: ExecutionContext = executor
+          val fDetails: Future[FilingDetail] = filingDetailsByPeriod(period, filingsActor, submissionActor)
+          onComplete(fDetails) {
+            case Success(filingDetails) =>
+              filingsActor ! Shutdown
+              submissionActor ! Shutdown
+              complete(ToResponseMarshallable(filingDetails))
+            case Failure(error) =>
+              filingsActor ! Shutdown
+              submissionActor ! Shutdown
+              complete(HttpResponse(StatusCodes.InternalServerError))
+          }
+        }
+      }
+    }
+
+  val submissionPath =
+    path("institutions" / Segment / "filings" / Segment / "submissions") { (fid, period) =>
+      val submissionsActor = system.actorOf(SubmissionPersistence.props(fid, period))
+      post {
+        val submissionsActor = system.actorOf(SubmissionPersistence.props(fid, period))
+        submissionsActor ! CreateSubmission
+        val fLatest = (submissionsActor ? GetLatestSubmission).mapTo[Submission]
```

Comment:
It's interesting, how this is implemented in the front-end mock server (which is somewhat mirrored w/ `GetLatestSubmission`) is that only the "current" submission is considered valid for API activity that includes a submission. Older submissions aren't explicitly marked as "canceled", they just aren't the latest and so are assumed de facto canceled. However, the front-end mock server modeled data a bit differently and didn't have a `submissionStatus` flag, here, we'd certainly need to adjust the `submissionStatus`, but I don't think a separate client request is necessary.

Then again, a separate client request would, in a simultaneous filers scenario, protect the in-progress work of the first user from the second-user's POSTing of a new submission... So I suppose I'm on the fence.